src/App.jsx

```diff
@@ -28,7 +33,7 @@ function App() {
 					/>
           <Route
 						path="/professionals"
-						element={<ProfessionalsPage />}
+						element={<ProfessionalsPage data={data} setData={setData} fullDataSet={fullDataSetRef.current} />}
 					/>
```

Comment:
The `.current` property is mutable and doesn't cause re-renders when changed. It behaves like instance variables in class components, holding onto state between renders without causing updates.